### DISCUSSION TITLE
"group by" performance

### DISCUSSION THREAD

**Discussion by JasonLi-cn** [databendlabs/databend#4600] [Q&A] [answered]
**Compare "group by" performance with ClickHouse**

**Question:**
**What configuration of Databend can I change to improve performance?**

CVM info:
- 84 processor (cat /proc/cpuinfo| grep "processor"| wc -l)
- Mem total: 314G

SQL:
`SELECT count(*) FROM numbers_mt(10000000) group by number limit 5;`

Databend:
<img width="791" alt="image" src="https://user-images.githubusercontent.com/23028310/160276677-7127b855-4187-45fe-ad3d-60e2e9dacbd9.png">

ClickHouse:
<img width="1115" alt="image" src="https://user-images.githubusercontent.com/23028310/160276689-2f79824c-7c8a-490d-9acc-498f0289afc4.png">

**Comment by sundy-li** [comment, marked as answer]
Yes, as the number of stored keys increases we did not have the two-level hashmap optimization like ClickHouse.
It can be enhanced in the future.

**Reply by sundy-li** [reply]
It's done in https://github.com/datafuselabs/databend/pull/8577
